src/main/java/org/cloudbus/cloudsim/brokers/DatacenterBrokerAbstract.java

```diff
@@ -78,6 +78,11 @@ public abstract class DatacenterBrokerAbstract extends CloudSimEntity implements
     private double failedVmsRetryDelay;
 
     /** @see #getVmFailedList() */
+
+    private int maxVmCreationRetries;
```

Comment:
Since multiple methods and attributes are related to this VM creation retry feature, creating a new class to move these attributes, constants (the DEF_CURRENT_VM_CREATION_RETRIES from DatacenterBroker) and methods avoid making the DatacenterBrokerAbstract even larger. This also ensures cohesion and separation of responsibilities.

The new class could be named as VmCreationRetry.

Reply:
Yes sure, will update this.